source/qx/io/jsonrpc/__init__.js

```diff
@@ -1,116 +0,0 @@
- *      transport.send(request);
- *      transport.send(notifiation);
- *      result = await request.getPromise();
- *    } catch(e) {
- *      // handle exceptions
- *    }
- *   })();
- * </pre>
- *
- * or using a batch, like so:
- *
- * <pre>
- *   (async()=>{
- *    const transport = new qx.io.jsonrpc.transport.Http("https://domain.com/endpoint");
- *    const batch = ()new qx.io.jsonrpc.message.Batch())
- *      .addRequest("method3")
- *      .addNotification("method4")
- *      .addRequest("method5","foo)
- *      .addRequest("method6", [{foo:"bar"}]);
- *    let results;
- *    try {
- *      transport.send(batch);
- *      results = await qx.Promise.all(batch.getPromises());
- *      // results will be an array with three items, the result of the requests
- *    } catch(e) {
- *      // handle exceptions
- *    }
- *   })();
- * </pre>
- *
- * 2) Using qx.io.jsonrpc.Client
- *
- * qx.io.jsonrpc.Client is an API built on top of the transports. It
- * will figure out the type of transport from the URI passed to it,
- * and provides a way to prepend a "service name" to all the method
- * names. This makes working with a JSON-RPC service a bit more
- * convenient, but is also less configurable, unless you subclass the
- * client and use a custom configuration behind the scenes (see below).
- *
- * Here is an example
- *
- * <pre>
- *   (async()=>{
- *    const client = new qx.io.jsonrpc.Client("https://domain.com/endpoint", "myServiceName");
- *    let result;
- *    try {
- *      client.sendNotification("other_method", [1,2,3]);
- *      result = await client.sendRequest("other_method", [1,2,3]);
- *    } catch(e) {
- *      // handle exceptions
- *    }
- *   })();
- * </pre>
- *
- * or using a batch:
- *
- * <pre>
- *   (async()=>{
- *    const client = new qx.io.jsonrpc.Client("https://domain.com/endpoint", "myServiceName");
- *    const batch = new qx.io.jsonrpc.message.Batch())
- *      .addRequest("method3")
- *      .addNotification("method4")
- *      .addRequest("method5","foo)
- *      .addRequest("method6", [{foo:"bar"}]);
- *    let results;
- *    try {
- *      results = await client.sendBatch(batch);
- *      // results will be an array with three items, the result of the requests
- *    } catch(e) {
- *      // handle exceptions
- *    }
- *   })();
- * </pre>
- *
- * If you want to use the client with a customized transport, create a class
- * that inherits from <code>qx.io.jsonrpc.transport.Abstract</code> or one of
- * its subclasses, override the methods which are needed to produce that custom
- * behavior (such as {@link qx.io.jsonrpc.transport.Http#_createRequest},
- * and provide a <code>defer</code> section which registers the behavior for
- * your particular class of URIs:
- *
- * <pre>
- * defer() {
- *   qx.io.jsonrpc.Client.registerTransport(/^http/, my.custom.Transport);
- * }
- * </pre>
- *
- * Afterwards, creating a client for alls urls
- * starting with "http" will use that custom behavior.
- *
```

Comment:
I'm not convinced that the Client class is necessary. It's not substantially easier to use than the first described method, and just adds more upkeep and questions to be answered.